static/app/components/searchQueryBuilder/hooks/useQueryBuilderState.tsx

```diff
@@ -859,22 +874,23 @@ export function useQueryBuilderState({
           };
         case 'COMMIT_QUERY':
           if (state.query === state.committedQuery) {
-            return state;
+            return {...state, previousAction: action.type};
           }
-          return {...state, committedQuery: state.query};
+          return {...state, committedQuery: state.query, previousAction: action.type};
         case 'UPDATE_QUERY': {
           const shouldCommitQuery = action.shouldCommitQuery ?? true;
 
           if (
             !hasWildcardOperators ||
-            !replaceRawSearchKeys ||
-            replaceRawSearchKeys.length === 0
+            !hasReplaceRawSearchKeys ||
+            state.previousAction === 'UPDATE_FREE_TEXT_ON_BLUR'
```

Comment:
This gives me some pause. It feels a little wrong to make this action dependent on actions that came before it (and I hate having to add additional state to the reducer for this one need).

Is there any way that we can call a separate action (or the same action with different parameters) from the free text component instead?